core/src/test/kotlin/ch/ergon/dope/buildTest/QueryBuilderTest.kt

```diff
@@ -787,4 +792,21 @@ class QueryBuilderTest {
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `should support selecting case`() {
+        val someBucket = someBucket()
+        val expected = "SELECT CASE `numberField` WHEN `other` THEN `numberField` END, " +
+            "CASE WHEN `booleanField` THEN `numberField` ELSE `stringField` END AS `alias` FROM `someBucket`"
+
+        val actual = create
+            .select(
+                case(someNumberField()).whenThen(someNumberField("other").resultsIn(someNumberField())),
```

Comment:
why is `case(someNumberField()).whenThen(someNumberField("other").resultsIn(2))`, not working?

Reply:
I noticed, that the extension functions for `resultsIn` used out insted of a Generic, so I changed it and your query should be valid.